src/color.rs

```diff
@@ -72,9 +72,70 @@ impl From<()> for OctColor {
     }
 }
 
+#[cfg(feature = "graphics")]
+impl From<BinaryColor> for OctColor {
+    fn from(b: BinaryColor) -> OctColor {
+        match b {
+            BinaryColor::On => OctColor::Black,
+            BinaryColor::Off => OctColor::White,
+        }
+    }
+}
+
+#[cfg(feature = "graphics")]
+impl From<OctColor> for embedded_graphics::pixelcolor::Rgb888 {
+    fn from(b: OctColor) -> Self {
+        let (r, b, g) = b.rgb();
+        Self::new(r, b, g)
+    }
+}
+
+#[cfg(feature = "graphics")]
+impl From<embedded_graphics::pixelcolor::Rgb888> for OctColor {
+    fn from(p: embedded_graphics::pixelcolor::Rgb888) -> OctColor {
+        use embedded_graphics::prelude::RgbColor;
+        let colors = [
+            OctColor::Black,
+            OctColor::White,
+            OctColor::Green,
+            OctColor::Blue,
+            OctColor::Red,
+            OctColor::Yellow,
+            OctColor::Orange,
+            OctColor::HiZ,
+        ];
+        // if the user has already mapped to the right color space, it will just be in the list
+        if let Some(found) = colors.iter().find(|c| c.rgb() == (p.r(), p.g(), p.b())) {
+            return *found;
+        }
+
+        // This is not ideal but just pick the nearest color
+        *colors
+            .iter()
+            .map(|c| (c, c.rgb()))
+            .map(|(c, (r, g, b))| {
+                let dist = ((r - p.r()) as u32).pow(2)
+                    + ((g - p.g()) as u32).pow(2)
+                    + ((b - p.b()) as u32).pow(2);
+                (c, dist)
+            })
+            .min_by_key(|(_c, dist)| *dist)
+            .map(|(c, _)| c)
+            .unwrap_or(&OctColor::White)
+    }
+}
+
+#[cfg(feature = "graphics")]
+impl From<embedded_graphics::pixelcolor::raw::RawU4> for OctColor {
+    fn from(b: embedded_graphics::pixelcolor::raw::RawU4) -> Self {
+        use embedded_graphics::prelude::RawData;
+        OctColor::from_nibble(b.into_inner()).unwrap()
+    }
+}
+
 #[cfg(feature = "graphics")]
 impl PixelColor for OctColor {
-    type Raw = ();
+    type Raw = embedded_graphics::pixelcolor::raw::RawU4;
```

Comment:
Did this change with the api or is this more efficient?

Reply:
It's actually just more correct, unrelated to API changes. It's basically stating that each `OctColor` pixel can be represented by 4 bits. This allows a raw buffer of 4 bit OctColor pixels to be readily converted into `ImageRaw`.
See [Raw Color Types Docs](https://docs.rs/embedded-graphics/0.7.1/embedded_graphics/pixelcolor/raw/index.html)

Reply:
:+1: